lib/Core/Site/Values/Content.php

```diff
@@ -143,4 +176,42 @@ private function buildField(array $properties = [])
         $this->fields[$field->fieldDefIdentifier] = $field;
         $this->fieldsById[$field->id] = $field;
     }
+
+    private function getLocations()
```

Comment:
For our last project we had to sort everything by Priority ASC, but the main point was that to make it consistent with the admin, but thats not the default sort for the admin either I think, so I think its too subjective.
I would vote for some simple sorting, dunno how your double level sort affects performance, but I guess its a bit overhead for eZ.
The best would be to make it customizable by a param, but then the `internalLocations` cache could cause problems withouth making that complex :)